flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/trait/FlinkRelDistribution.scala

```diff
@@ -119,7 +119,7 @@ class FlinkRelDistribution private (
           try {
             val i = mapping.getTargetOpt(fieldCollation.getFieldIndex)
             if (i >= 0) {
-              newFieldCollations.add(fieldCollation.copy(i))
```

Comment:
deprecated at https://github.com/apache/calcite/commit/513f4d2510a47e8d0b55f35dac56785c16fa7d5f
in favor of `withXXX` methods